deploy/crds/integreatly_v1alpha1_postgres_cr.yaml

```diff
@@ -11,3 +11,10 @@ spec:
   tier: development
   # i want a postgres storage for the type workshop
   type: workshop
+  labels:
```

Comment:
Comment around how we are doing this. Do we need to add this info to the CR. Can we not generate these dynamically? 
I would lean toward not having them added to the CR, but think maybe we should have more of a discussion around this

Reply:
Would be pretty nice to be able to tag based on flexible list of key/value pairs, that way we aren't adding labels that are very integreatly specific. 

For integration with integreatly we could have a cro util function that accepts a map of key value pairs, and updates the cr dynamically with the labels. If labels exist on the cr we can proceed to tag the aws resources.

Reply:
For `env` I think this could be an `env var`, if set then we label with the organisation. If not we ignore this label. 
For the other label, I know we need to be able to list resources from cluster X and product Y. So the product may need to be set externally in the CR, as I can't see how CRO will know what product the resource belongs too. Unless can we some how get the product name from the name passed to the CR?

Reply:
Yeah we'll need to pass it in from integreatly. My idea is we have a map of labels in the integreatly product, something like :
```go
labels := &map[string]string{
     "productName": "3scale",
     "clusterId": clusterId,
     ...
}
```

Then when we create the cro resource in integreatly, we label the cr in the modification function:
```go
postgres, err := croUtil.ReconcilePostgres(ctx, client, r.installation.Spec.Type, tier, postgresName, ns, postgresName, ns, func(cr metav1.Object) error {
	// add the labels
	cr.Spec.Labels = labels 

	// add the owner
	ownerutil.EnsureOwner(cr, r.installation)
	return nil
})
```

In cro aws providers, if `cr.Spec.Labels != nil`, we can loop through the map and dynamically add the `key`/`value` pairs 

```go
tags := []*elasticache.Tag{}
for k, v := range cr.Spec.Labels {
    tag := *elasticache.Tag{
        Key: fmt.Sprintf("%s/%s", org, k),
        Value: v,
    }
    tags = append(tags, tag)
}
```

I haven't tried it myself obviously, but I don't see why it wouldn't work

Reply:
We could also define a struct in the integreatly operator with fixed fields like `ProductName`, `ClusterID`, etc, so that we enforce a standard for key names. We can then have a util function there too which dumps the struct values into a map that we can use to label the cr

Reply:
Adding the productName to the CR was outlined in the jira https://issues.redhat.com/browse/INTLY-3968

env , I have no objection to it being a type omit on empty, I think there was a need for it to be an environment variable, and this seemed to be best practice https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/ to set it in the spec but maybe it can be add to another spec rather that the CR

There is this ticket in the backlog for integreatly integration of tagging, https://issues.redhat.com/browse/INTLY-4377, it may be the best place for a discussion. 

ClusterID is got dynamically in CRO so don't need to pass it from Integreatly in the CR.
The operator-sdk seems to give out about `&map[string]string` types as an api violation, But I didn't play around with it too much after the first error.

Reply:
Reading that Jira again, it seems like the labels should be added to the `metadata.labels` field in the custom resource: 

```yml
metadata:
  labels:
    integreatly.org/product-name: 3scale
```

I think that's actually really nice because it won't require any breaking changes to the CRD, so it  also won't require as much refactoring on the integreatly side. 

I also personally would be a fan of leaving the env out because it could be easily done as a variable on the integreatly side as well. Not sure what you guys think?